File: templates/python/starter.py
#!/usr/bin/env python3
"""
Starter template for Python 3.11+
Entrypoint: Run with `python3 starter.py`
"""


def main():
    """
    Main entry point.
    Sample I/O:
      Input: (via stdin or args)
      Output: (via stdout)
    """
    # Read input
    user_input = input("Enter your name: ")
    
    # Process
    result = f"Hello, {user_input}!"
    
    # Write output
    print(result)


if __name__ == "__main__":
    main()

File: templates/rust/starter.rs
use std::io::{self, Write};

/*
Starter template for Rust (stable)
Entrypoint: Run with `rustc starter.rs && ./starter`
Or with cargo: Create a new project with `cargo new myproject`, replace src/main.rs,
               then run with `cargo run`
*/

fn main() {
    /*
    Sample I/O:
      Input: (via stdin or args)
      Output: (via stdout)
    */

    // Read input
    print!("Enter your name: ");
    io::stdout().flush().unwrap();
    
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");
    
    let input = input.trim();

    // Process
    let result = format!("Hello, {}!", input);

    // Write output
    println!("{}", result);
}
